File: backend/app/schemas/share.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field


class CreateShareRequest(BaseModel):
    type: Literal["text", "file"]
    encrypted_payload: str = Field(min_length=1)
    file_name: str | None = None
    file_size: int | None = Field(default=None, ge=0)
    mime_type: str | None = None
    expires_in: int = Field(gt=0)
    burn_after_read: bool = False


class CreateShareResponse(BaseModel):
    id: str


class GetShareResponse(BaseModel):
    id: str
    type: Literal["text", "file"]
    encrypted_payload: str
    file_name: str | None
    file_size: int | None
    mime_type: str | None
    burn_after_read: bool
    expires_at: datetime

File: frontend/src/types/share.ts
export type ShareType = 'text' | 'file'

export type CreateShareFormValues = {
  type: ShareType
  content: string
  file: File | null
  expires_in: number
  burn_after_read: boolean
}

export type EncryptedPayload = {
  iv: string
  ciphertext: string
}

export type CreateSharePayload = {
  type: ShareType
  encrypted_payload: string
  file_name?: string | null
  file_size?: number | null
  mime_type?: string | null
  expires_in: number
  burn_after_read: boolean
}

export type CreateShareResponse = {
  id: string
}

export type GetShareResponse = {
  id: string
  type: ShareType
  encrypted_payload: string
  file_name: string | null
  file_size: number | null
  mime_type: string | null
  burn_after_read: boolean
  expires_at: string
}
